File: historyscan-back/src/main/java/io/sch/historyscan/domain/contexts/codebase/find/FindCodeBases.java
package io.sch.historyscan.domain.contexts.codebase.find;

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureAPI;

import java.util.List;

@HexagonalArchitectureAPI
@FunctionalInterface
public interface FindCodeBases {
    List<CurrentCodeBase> fromDisk();
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/domain/contexts/codebase/find/FindCodeBases.kt
package io.sch.historyscan.domain.contexts.codebase.find

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureAPI

@HexagonalArchitectureAPI
fun interface FindCodeBases {
    fun fromDisk(): List<CurrentCodeBase>
}
